edgedb/retry.py

```diff
@@ -37,11 +39,17 @@ async def __aexit__(self, extype, ex, tb):
             # to propagate it to cancel the whole task.
             # NOTE: rollback error is always swallowed, should we use
             # on_log_message for it?
+        except ConnectionError:
+            reconnect = True
+
+        if reconnect:
+            await self._owner._reconnect()
 
-        if (
-            extype is not None and
+        if extype is not None and (
             issubclass(extype, errors.EdgeDBError) and
             ex.has_tag(errors.SHOULD_RETRY)
+        ) or (
+            issubclass(extype, ConnectionError)
```

Comment:
If I understood this correctly - we want to `_reconnect()` on all ConnectionErrors, but only retry on the ones that is not from `commit()`? A comment would be super helpful 🙏

Reply:
We don't retry on network errors during `COMMIT`, because there is no way to know if it actually reached the server or not.  Also, there's no point in retrying `ROLLBACK`, since its issuance already indicated a non-retryable condition.

Reply:
changed